test/integration/puppetlabs/services/master/master_service_test.clj

```diff
@@ -440,3 +461,212 @@
+                                               (.runScriptlet
+                                                container
+                                                ;; So long as the server and port are accessible, it
+                                                ;; doesn't actually matter whether the endpoint is
+                                                ;; reachable or not - we just need to make requests
+                                                ;; with the given metric ids.
+                                                (format "$c.get('/fake', {}, {:metric_id => %s})"
+                                                        metric-id-as-string)))]
+             (testing "http-client-metrics key in master status"
+               (make-request-with-metric-id "['foo', 'bar', 'baz']")
+               (make-request-with-metric-id "['hello', 'cruel', 'world']")
+               (make-request-with-metric-id "['puppetdb', 'query']")
+               (make-request-with-metric-id "['fake', 'fake', 'fakery']")
+               (let [metric-ids (map :metric-id (get-http-client-metrics-status))]
+
+                 (testing "has metric-ids from master service"
+                   (is (some #(= ["puppetdb" "query"] %) metric-ids)))
+
+                 (testing "has metric-ids added by `add-metric-ids-to-http-client-metrics-list`"
+                   (is (some #(= ["foo" "bar"] %) metric-ids))
+                   (is (some #(= ["hello" "cruel" "world"] %) metric-ids)))
+
+                 (testing "doesn't include metrics for metric-ids it is not supposed to have"
+                   (is (not-any? #(= ["fake" "fake" "fakery"] %) metric-ids)))
+
+                 (testing "has correct set of metric-ids"
+                   (is (= #{["foo" "bar"] ["hello" "cruel" "world"] ["puppetdb" "query"]}
+                          (set metric-ids)))))))
+           (finally
+             (jruby-testutils/return-instance jruby-service jruby-instance :add-metric-ids-test))))))))
+
+(deftest ^:integration http-report-processor-client-metrics-test
+  (testing "HTTP client metrics from the http report processor are added to status"
+    (bootstrap-testutils/with-puppetserver-running
+     app
+     {:jruby-puppet {:max-active-instances 2 ; we need 2 jruby-instances since processing the report uses an instance
+                     :master-code-dir test-resources-code-dir
+                     :master-conf-dir master-service-test-runtime-dir}
+      :metrics {:server-id "localhost"}}
+     (let [jruby-service (tk-app/get-service app :JRubyPuppetService)
+           jruby-instance (jruby-testutils/borrow-instance jruby-service :http-report-processor-metrics-test)
+           container (:scripting-container jruby-instance)]
+       (try
+         ;; we don't care at all about the content of the report, just that it is valid
+         (let [report (.runScriptlet container "Puppet::Transaction::Report.new('apply').to_pson")]
```

Comment:
Seems like to_json for the request payload might be fine here already.  Since I don't think this would produce any funky character encoding that would require it to be rendered as "pson", maybe using the "Content-Type" of "text/pson" for the request but a request body that's actually json would be okay?  If we need to wait on the json "Content-Type" to work until after [PUP-7259](https://tickets.puppetlabs.com/browse/PUP-7259) comes in, though, maybe it would be better to just use to_pson here for now and change it later?